aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -286,8 +286,12 @@ protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
   }
 
   @Override
-  public List<TableIdentifier> listTables(Namespace namespace) {
-    namespaceExists(namespace);
+  public List<TableIdentifier> listTables(Namespace namespace) throws NoSuchNamespaceException {
+    // check if namespace exists
+    if (!namespaceExists(namespace)) {
+      throw new NoSuchNamespaceException(
+          "Cannot list tables of namespace %s because it does not exist", namespace);
+    }
```

Comment:
This is a behavior change but going through the Java doc, this is the expectation for catalog impls and it does make sense. I guess somehow this just got missed originally.

```
  /**
   * Return all the identifiers under this namespace.
   *
   * @param namespace a namespace
   * @return a list of identifiers for tables
   * @throws NoSuchNamespaceException if the namespace is not found
   */
```